include/net/aws_iot.h

```diff
@@ -44,7 +46,9 @@ enum aws_iot_topic_type {
 	 *  $aws/things/<thing-name>/shadow/delete, publishing an empty message
 	 *  to this topic deletes the device Shadow document.
 	 */
-	AWS_IOT_SHADOW_TOPIC_DELETE
+	AWS_IOT_SHADOW_TOPIC_DELETE,
+	/** Use when publishing to a topic not predefined above. */
+	AWS_IOT_APPLICATION_TOPIC = 0x0
```

Comment:
I think this could be moved to the top of the list and comment changed accordingly, so that the type values are in ascending order. Then in next release, `AWS_IOT_SHADOW_TOPIC_UNKNOWN` can be removed